File: c/lab01/mario.c
#include <stdio.h>

int get_height(void);

int main(void)
{
    int h;

    h = get_height();

    for (int i = 0; i < h; i++)
    {
        for (int j = 0; j < h - i; j++)
        {
            printf(" ");
        }

        for (int k = 0; k < i; k++)
        {
           printf("#");
        }

        printf(" ");

        for (int l = 0; l < i; l++)
        {
           printf("#");
        }
        
        printf("\n");
    }
}

int get_height(void)
{
    int i;

    do 
    {
        printf("Height: ");
        scanf("%d", &i);
    } while (i < 1 || i > 8);
    return i;
}

File: python/problemset6/mario.py
def main():
    height = get_height()

    for i in range(1, height + 1):
        print(" " * (height - i), end="")

        print("#" * i, end="")

        print("  ", end="")

        print("#" * i)


def get_height():
    while True:
        try:
            height = int(input("Height: "))

            if 1 <= height <= 8:
                return height
            else:
                print("Height must be between 1 and 8")
        except ValueError:
            print("Not an integer")


main()
